### DISCUSSION TITLE
Dependency error when adding apache-airflow-providers-microsoft-mssql

### DISCUSSION THREAD

**Discussion by leanst-daniel** [apache/airflow#32963] [General] [answered]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Hi all,

when I redeployed our testing Airflow environment I ran into the following error:

> 
> Defaulting to user installation because normal site-packages is not writeable
> Collecting apache-airflow-providers-microsoft-mssql==3.3.2
>   Downloading apache_airflow_providers_microsoft_mssql-3.3.2-py3-none-any.whl (16 kB)
> Requirement already satisfied: apache-airflow-providers-common-sql>=1.3.1 in /home/airflow/.local/lib/python3.9/site-packages (from apache-airflow-providers-microsoft-mssql==3.3.2) (1.3.4)
> Requirement already satisfied: apache-airflow>=2.3.0 in /home/airflow/.local/lib/python3.9/site-packages (from apache-airflow-providers-microsoft-mssql==3.3.2) (2.5.2)
> Collecting pymssql>=2.1.5
>   Downloading pymssql-2.2.8.tar.gz (171 kB)
>      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 171.5/171.5 kB 88.0 MB/s eta 0:00:00
>   Installing build dependencies: started
>   Installing build dependencies: finished with status 'done'
>   Getting requirements to build wheel: started
>   Getting requirements to build wheel: finished with status 'error'

We add the pip requirement "apache-airflow-providers-microsoft-mssql==3.3.2". As soon as I remove the extra pip package everything works fine.

I'm not sure since when this issue exsists but our last deployment was on the 28th June.#

best regards

### How to reproduce

Add apache-airflow-providers-microsoft-mssql==3.3.2 to pip requirements

### Operating System

Docker airflow:2.5.2-python3.9

### Versions of Apache Airflow Providers

apache-airflow-providers-microsoft-mssql==3.3.2

### Deployment

Other Docker-based deployment

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Reply by potiuk** [reply]
You need to separately install pymssql in your environment (I guess you are using ARM). This is a problem with pymssql not working with the new CPython 3.0.0 released recently and it is a pymssql problem, not Airflow - other than providing workaround - see https://github.com/apache/airflow/issues/32672#issuecomment-1647007726 

In your case likely you need to add several steps in your dockerfile to generate constraints file with the right version of Cpython, set the environment variable and run `pip install pymssql`. 

The reasons why it is what it is are explained here:
https://github.com/apache/airflow/issues/32672#issuecomment-1647022756 . We cannot do much - we can just wait until Microsoft releases pymssql version that is compatible.

**Reply by potiuk** [reply]
Ah - I see that https://github.com/pymssql/pymssql/pull/803 has been merged today and new pymssql 2.2.8 compatible with Cpython 3 has been released. So I can actually fix some of the constraints for some of te past versions of airflow to make it works without the workaround

**Reply by potiuk** [reply, marked as answer]
FYI. I proposed we will update constraints https://lists.apache.org/thread/gcctmgrfysgybqvf3s40oxn4z6gftbrq -> once PyMSSQL will add binary packages for linux (they are missing in 2.2.8 for that) and lazy consensus will be reached, I will update constraints to make it works wwithout the workaround. In the meantime I **think** if you you should be able to install the provider with `pymssql==2.2.8` without constraints and it **should** work.

**Reply by leanst-daniel** [reply]
Awesome, thank you! I'll try it later and let you know

**Reply by leanst-daniel** [reply]
@potiuk Adding pymssql==2.2.8 to my pip requirements worked. Thank you!
